### PR TITLE
Move showCollection to collection.js

### PR DESCRIPTION
**Pull request by o5v3**
Quick restructure to take away some size from cards.js and separate its concerns from the collection page. From what I see in the other current PRs this shouldn't produce conflicts with their code.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -66,6 +67,30 @@ function toggleCollection(event) {
     toggle === "true" ? "Hide Full Collection" : "Show Full Collection";
 }
 
+function setupDropdown(dropdown, storage_key, default_value) {
```
Comment (miffens) [inline js/main.js]: (nit) use camel case for the arguments (storageKey). Snake case (or whatever storage_key is) feels suitable for "backend data" (cards_data)

File: js/collection.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  } else {
+    control.classList.remove("hidden");
+    control.onclick = onclick;
+  }
+}
+
+//If we will show more cards than the amount specified in cards_per_page,
+//then we split them into parts and show the pagination controls.
+//Note: If cards_per_page is set to "all", the comparison casts it to NaN, and as such it will always return false.
+function paginateCards(cards, cards_per_page, page, navigationFunction) {
```
Comment (miffens) [inline js/collection.js:98]: could you comment the shape of what this function returns and a summary of what the function does, ie
"Returns cards_per_page number of card data objects to render on the specified page. Additionally, wires and shows/ hides the pagination controls accordingly"
Took me a while to figure this out :')

File: js/collection.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    //If we don't need pagination, we hide the controls.
+    PAGINATION_BTNS.container.classList.add("hidden");
+    return cards;
+  }
+}
+
+export function showCollection(cards_data, htmlLocation, page = 1) {
+  let sort = localStorage.getItem("sort") ?? "Collector Number";
+  let cards_per_page = localStorage.getItem("page-size") ?? 10;
+  let fullCollection = localStorage.getItem("fullCollection") === "true";
```
Comment (miffens) [inline js/collection.js]: (nit) I like naming booleans as a question to a yes/no answer: `showFullCollection`? yes/ no
`fullCollection` makes me think it's an array of all the cards

File: js/cards.js [hunk 4]
```diff
@@ -91,7 +46,7 @@ export async function defineCardComponent() {
     //Returns an url of the form:
     //`https://res.cloudinary.com/${CLOUD_NAME}/image/upload/${RARITY}/${FILENAME}.png`
     getImageURL() {
-      return `${CLOUDINARY_URL}/q_auto/${this.data["Rarity Folder"]}/${this.data["Filename"]}.png`;
+      return `${CLOUDINARY_URL}q_auto/${this.data["Rarity Folder"]}/${this.data["Filename"]}.png`;
     }
 
     setupCardForGacha() {
```
Comment (miffens) [inline js/cards.js:52]: The gacha specific code might be nice to split out too to be consistent with the collection.js organization. We can do it later if we have time

**Review by miffens — APPROVED** [review]

File: js/main.js (hunk 1 above)
Comment (o5v3) [reply js/main.js]: Done. o7

File: js/collection.js (hunk 2 above)
Comment (o5v3) [reply js/collection.js:98]: Done o7. The function looks more complex than it really is, but hopefully the comments help in clarifying how it works.

File: js/collection.js (hunk 3 above)
Comment (o5v3) [reply js/collection.js]: Done o7. Changed it across all files too.

File: js/cards.js (hunk 4 above)
Comment (o5v3) [reply js/cards.js:52]: Agreed, having page-specific code in its own file is better for organization. I'll focus on some items in the Notion board and, if time allows, I'll make the split. o7

**Pull request merged**